### PR TITLE
(PostCSS plugin): Add `postcss` to  used dependencies when using PostCSS with Tailwind CSS

### PR DESCRIPTION
**Pull request by taro-28**
Closes https://github.com/webpro-nl/knip/issues/762

### CODE REVIEW

File: packages/knip/src/types/plugins.ts [hunk 1]
```diff
@@ -33,7 +33,8 @@ export type ResolveEntryPaths<T = any> = (config: T, options: PluginOptions) =>
 // biome-ignore lint/suspicious/noExplicitAny: TODO
 export type ResolveConfig<T = any> = (config: T, options: PluginOptions) => Promise<string[]> | string[];
 
-export type Resolve = (options: PluginOptions) => Promise<string[]> | string[];
+// biome-ignore lint/suspicious/noExplicitAny: TODO
+export type Resolve<T = any> = (config: T, options: PluginOptions) => Promise<string[]> | string[];
```
Comment (taro-28) [inline packages/knip/src/types/plugins.ts:37]: I added `config` as the first parameter to align the function's format with `ResolveEntryPaths` and `ResolveConfig`.

**Comment by taro-28** [comment]
Checking for failed tests now🙏

File: packages/knip/src/plugins/postcss/index.ts [hunk 2]
```diff
@@ -27,10 +27,19 @@ const resolveConfig: ResolveConfig<PostCSSConfig> = config => {
     : [];
 };
 
+const resolve: Resolve<PostCSSConfig> = async (options, config) => {
+  for (const plugin of await resolveConfig(options, config)) {
+    // Because postcss is not included in peerDependencies of tailwindcss
+    if (plugin === 'tailwindcss') return ['postcss'];
```
Comment (webpro) [inline packages/knip/src/plugins/postcss/index.ts:33]: What's the reason for not adding a check for `tailwindcss` in `config.plugins` to the `resolveConfig` function directly?

**Review by webpro — COMMENTED** [review]
Thanks for the PR @taro-28! Appreciated. I do have a question, though.

File: packages/knip/src/plugins/postcss/index.ts (hunk 2 above)
Comment (taro-28) [reply packages/knip/src/plugins/postcss/index.ts:31]: I misunderstood the implementation policy🙏

FIxed!
https://github.com/webpro-nl/knip/pull/764/commits/03a7583d1f1b96d4a7b1b8675d72b09882874c55

Comment (taro-28) [reply packages/knip/src/plugins/postcss/index.ts:31]: Sorry, there was a mistake.

Comment (webpro) [reply packages/knip/src/plugins/postcss/index.ts:31]: Sorry taro. Is there anything in the documentation or code that could have prevented this? Maybe we can improve it for the next contributor.

Thanks for the fix!

Comment (taro-28) [reply packages/knip/src/plugins/postcss/index.ts]: > Sorry taro. Is there anything in the documentation or code that could have prevented this? Maybe we can improve it for the next contributor.

I re-read the [documentation](https://knip.dev/guides/writing-a-plugin#6-resolveconfig) again, and it succinctly describes how to use `resolveConfig`, so I thought it was fine as it is.

My misunderstanding was due to overcomplicating the issue, as it was a slightly exceptional modification, such as adding the plugin's dependency itself (postcss).

File: packages/knip/src/plugins/postcss/index.ts [hunk 3]
```diff
@@ -18,13 +18,21 @@ const config = [
...
       })
     : [];
+
+  for (const plugin of plugins) {
+    // Because postcss is not included in peerDependencies of tailwindcss
+    if (plugin === 'tailwindcss') {
+      return [...plugins, 'postcss'];
+    }
+  }
+  return plugins;
```
Comment (webpro) [inline packages/knip/src/plugins/postcss/index.ts:35]:
```suggestion
  return plugins.includes('tailwindcss') ? [...plugins, 'postcss'] : plugins;
```

**Comment by webpro** [comment]
Thank you! 🙏

**Comment by webpro** [comment]
:rocket: _This pull request is included in v5.27.3. See [Release 5.27.3](https://github.com/webpro-nl/knip/releases/tag/5.27.3) for release notes._

_Using Knip in a commercial project? Please consider [becoming a sponsor](https://knip.dev/sponsors)._

File: packages/knip/test/plugins/postcss-tailwindcss.test.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+const cwd = resolve('fixtures/plugins/postcss-tailwindcss');
+
+test('Find dependencies with the PostCSS plugin (with tailwindcss)', async () => {
+  const { issues, counters } = await main({
+    ...baseArguments,
+    cwd,
+  });
+
+  assert(issues.unlisted['postcss.config.js']['tailwindcss']);
```
Comment (Crustan) [inline packages/knip/test/plugins/postcss-tailwindcss.test.ts:16]: Should this also support postcss.config.mjs?

**Pull request merged**